haystack_experimental/components/splitters/hierarchical_doc_splitter.py

```diff
@@ -19,10 +19,10 @@ class HierarchicalDocumentSplitter:
     ## Usage example
     ```python
     from haystack import Document
-    from haystack.components.builders import HierarchicalDocumentBuilder
+    from haystack_experimental.components.splitters import HierarchicalDocumentSplitter
 
     doc = Document(content="This is a simple test document")
-    builder = HierarchicalDocumentBuilder(block_sizes=[3, 2], split_overlap=0, split_by="word")
+    builder = HierarchicalDocumentSplitter(block_sizes=[3, 2], split_overlap=0, split_by="word")
```

Comment:
Will probably be more clear this way?
```suggestion
    splitter = HierarchicalDocumentSplitter(block_sizes=[3, 2], split_overlap=0, split_by="word")
```